typings.d.ts

```diff
@@ -271,24 +307,39 @@ declare module 'contentful-ui-extensions-sdk' {
     invocation?: Object;
   }
 
-  export interface ExtensionSDK {
-    field?: FieldAPI;
+  export interface BaseExtensionSDK {
+    /** Allows to read and update the value of any field of the current entry and to get the entry's metadata */
     entry: EntryAPI;
+    /** Information about the content type of the entry. */
     contentType: ContentType;
+    /** Exposes methods that allow the extension to read and manipulate a wide range of objects in the space. */
     space: SpaceAPI;
-    locales: LocalesAPI;
+    /** Information about the current user and roles */
     user: User;
+    /** Information about the current locales */
+    locales: LocalesAPI;
+    /** Methods to update the size of the iframe the extension is contained within.  */
     window: WindowAPI;
+    /** Methods for opening UI dialogs: */
     dialogs: DialogsAPI;
+    /** Methods for navigating between entities stored in a Contentful space. */
     navigator: NavigatorAPI;
+    /** Methods for displaying notifications. */
     notifier: NotifierAPI;
+    /** Exposes extension configuration parameters */
     parameters: ParametersAPI;
     location: LocationAPI;
   }
 
-  type InitCallback = (sdk: ExtensionSDK) => void;
+  export type SidebarExtensionSDK = BaseExtensionSDK;
+
+  export type FieldExtensionSDK = BaseExtensionSDK & {
+    /** Gives you access to the value and metadata of the field the extension is attached to. */
+    field: FieldAPI
+  }
+
+  export const init: (initCallback: (sdk: BaseExtensionSDK | FieldExtensionSDK | SidebarExtensionSDK) => any) => void;
```

Comment:
There's also `DialogExtensionSDK` with `close` ;)

Reply:
oh, good to know. I'll add it.

Reply:
done!